tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/rest/TagServiceEntityProvider.java

```diff
@@ -215,6 +221,42 @@ public JSONObject getTagsPaginatedByPrefixInLabel(EntityView view, Map<String, O
         }
     }
 
+    @EntityCustomAction(action = "getTagsByAssignmentId", viewKey = EntityView.VIEW_LIST)
+    public JSONObject getTagsByAssignmentId(EntityView view, Map<String, Object> params) {
+        try {
+            WrappedParams wp = new WrappedParams(params);
+            String assignmentId = wp.getString("assignmentId");
+
+            Assignment assignment = assignmentService.getAssignment(assignmentId);
+            String siteId = assignment.getContext();
```

Comment:
SiteId should be a param , this can be called from other tools too. Method should be renamed too